Return type of public static method from exported class has or is using private name 'BaseWithPlugins'

running `tsc index.ts --declaration` currently throws an error

```
> tsc index.ts --declaration

index.ts:25:10 - error TS4052: Return type of public static method from exported class has or is using private name 'BaseWithPlugins'.

25   static plugin<T extends TestPlugin | TestPlugin[]>(plugin: T) {
            ~~~~~~
```

I'm not sure how to fix that problem yet, help much appreciated!

--- a/index.ts
+++ b/index.ts
@@ -25,9 +25,9 @@ export class Base {
   static plugin<T extends TestPlugin | TestPlugin[]>(plugin: T) {
     const currentPlugins = this.plugins;
 
-    class BaseWithPlugins extends this {
+    const BaseWithPlugins = class extends this {
       static plugins = currentPlugins.concat(plugin);
-    }
+    };
 
     type Extension = ReturnTypeOf<T>;
     return BaseWithPlugins as typeof BaseWithPlugins & Constructor<Extension>;

